packages/reactive-element/src/test/decorators/queryAssignedElements_test.ts

```diff
@@ -0,0 +1,195 @@
+    footerAssigned!: Element[];
+
+    @queryAssignedElements({slotName: 'footer', flatten: false})
+    footerNotFlattenedSlot!: Element[];
+
+    @queryAssignedElements({
+      slotName: 'footer',
+      flatten: true,
+      selector: '.item',
+    })
+    footerAssignedFiltered!: Element[];
+
+    override render() {
+      return html`
+        <slot></slot>
+        <slot name="footer"></slot>
+      `;
+    }
+  }
+  customElements.define('assigned-elements-el', D);
+
+  class E extends RenderingElement {
+    @queryAssignedElements() defaultAssigned!: Element[];
+
+    @queryAssignedElements({slotName: 'header'}) headerAssigned!: Element[];
+
+    override render() {
+      return html`
+        <slot name="header"></slot>
+        <slot></slot>
+      `;
+    }
+  }
+  customElements.define('assigned-elements-el-2', E);
+
+  const defaultSymbol = Symbol('default');
+  const headerSymbol = Symbol('header');
+  class S extends RenderingElement {
+    @queryAssignedElements() [defaultSymbol]!: Element[];
+
+    @queryAssignedElements({slotName: 'header'}) [headerSymbol]!: Element[];
+
+    override render() {
+      return html`
+        <slot name="header"></slot>
+        <slot></slot>
+      `;
+    }
+  }
+  customElements.define('assigned-elements-el-symbol', S);
+
+  // Note, there are 2 elements here so that the `flatten` option of
+  // the decorator can be tested.
+  class C extends RenderingElement {
+    div!: HTMLDivElement;
+    div2!: HTMLDivElement;
+    div3!: HTMLDivElement;
+    assignedEls!: D;
+    assignedEls2!: E;
+    assignedEls3!: S;
+    @queryAssignedElements() missingSlotAssignedElements!: Element[];
+
+    override render() {
+      return html`
+        <assigned-elements-el
+          ><div id="div1">A</div>
+          <slot slot="footer"></slot
+        ></assigned-elements-el>
+        <assigned-elements-el-2><div id="div2">B</div></assigned-elements-el-2>
+        <assigned-elements-el-symbol
+          ><div id="div3">B</div></assigned-elements-el-symbol
+        >
+      `;
+    }
+
+    override firstUpdated() {
+      this.div = this.renderRoot.querySelector('#div1') as HTMLDivElement;
+      this.div2 = this.renderRoot.querySelector('#div2') as HTMLDivElement;
+      this.div3 = this.renderRoot.querySelector('#div3') as HTMLDivElement;
+      this.assignedEls = this.renderRoot.querySelector(
+        'assigned-elements-el'
+      ) as D;
+      this.assignedEls2 = this.renderRoot.querySelector(
+        'assigned-elements-el-2'
+      ) as E;
+      this.assignedEls3 = this.renderRoot.querySelector(
+        'assigned-elements-el-symbol'
+      ) as S;
+    }
+  }
+  customElements.define(generateElementName(), C);
+
+  setup(async () => {
+    container = document.createElement('div');
+    container.id = 'test-container';
+    document.body.appendChild(container);
+    el = new C();
+    container.appendChild(el);
+    await el.updateComplete;
+    await el.assignedEls.updateComplete;
```

Comment:
Use Promise.all() and await all the elements? Or await a setTimeout?

Reply:
These need to be sequential so opted into using `setTimeout`.